libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx

```diff
@@ -0,0 +1,437 @@
+    workspace?: string;
+    filter?: IAttributeFilter;
+    identifier?: string;
+    title?: string;
+    selectedItems?: Array<Partial<IAttributeElement>>;
+    onApply: (filter: IAttributeFilter, isInverted: boolean) => void;
+    onError: (error: any) => void;
+}
+
+/**
+ * @public
+ */
+export type IAttributeFilterButtonProps = IAttributeFilterButtonOwnProps & WrappedComponentProps;
+
+const DefaultFilterError: React.FC = injectIntl(({ intl }) => {
+    const text = intl.formatMessage({ id: "gs.filter.error" });
+    return <div className="gd-message error s-button-error">{text}</div>;
+});
+
+const DropdownBody: React.FC<{
+    isMobile?: boolean;
+    closeDropdown?: () => void;
+    children: React.ReactElement<any>;
+}> = (props) => {
+    return props.children;
+};
+
+const DropdownButton: React.FC<{
+    isMobile?: boolean;
+    isOpen?: boolean;
+    title: string;
+    subtitle: string;
+}> = ({ isMobile, isOpen, title, subtitle }) => {
+    return (
+        <div
+            className={cx("gd-attribute-filter-button", {
+                "is-active": isOpen,
+                "gd-attribute-filter-button-mobile": isMobile,
+            })}
+        >
+            <div className="button-content">
+                <div className="button-title">{title}</div>
+                <div className="button-subtitle">{subtitle}</div>
+            </div>
+        </div>
+    );
+};
+
+const LIMIT = MAX_SELECTION_SIZE + 50;
+
+export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> = (props) => {
+    const [validElements, setValidElements] = useState<IElementQueryResultWithEmptyItems>(null);
+    const [selectedFilterOptions, setSelectedFilterOptions] = useState<Array<IAttributeElement>>([]);
+    const [isInverted, setIsInverted] = useState(true);
+    const [prevSelectedFilterOptions, setPrevSelectedFilterOptions] = useState<Array<IAttributeElement>>([]);
+    const [prevIsInverted, setPrevIsInverted] = useState(true);
+    const [firstLoad, setFirstLoad] = useState(true);
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState<any>(null);
+    const [searchString, setSearchString] = useState("");
+    const [offset, setOffset] = useState(0);
+    const [limit, setLimit] = useState(LIMIT);
+    const [totalCount, setTotalCount] = useState(LIMIT);
+    const [filterOptions, setFilterOptions] = useState<Array<IAttributeElement>>([]);
+    const [title, setTitle] = useState("");
+    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+    const dropdownRef = useRef<Dropdown>(null);
+
+    useEffect(() => {
+        getElementTotalCount();
+        loadAttributeTitle(false);
+    }, []);
+
+    useEffect(() => {
+        setValidElements(null);
+        setError(null);
+        setIsLoading(false);
+        setOffset(0);
+        setLimit(LIMIT);
+
+        getElements();
+    }, [props.workspace, searchString]);
+
+    const loadAttributeTitle = async (force = false) => {
```

Comment:
Consider using `useCancellablePromise` for this, it could save you some of the explicit orchestration.